### ISSUE TITLE
[LightGBM] [Warning] Unknown parameter: linear_tree

### ISSUE THREAD

**Issue by flippercy** [microsoft/LightGBM#3910]
I am using lightgbm via reticulate in R:

parameters = {
    'objective': 'binary',
    'linear_tree': 'true',
    'tree_learner': 'serial',
    'metric': 'auc',
    'n_jobs': 12,
    'seed': 19850606,
    'boosting': 'gbdt',
    'num_leaves': 31,
    'learning_rate': 0.05,
    'verbose': 1,
    'monotone_constraints': lightGBM_monotoneVector,
    'monotone_constraints_method':'advanced'
    
}

lightgbm_model = lgb.train(parameters,
                       train_data,
                       valid_sets=test_data,
                       num_boost_round=500,
                       early_stopping_rounds=50)

However, I got a warning:

[LightGBM] [Warning] Unknown parameter: linear_tree

What is the reason? I know that linear_tree is not available in the R library of lightGBM but here I am using the python package via reticulate.

Thank you.

**Comment by StrikerRUS** [comment]
Hi @flippercy !

Linear trees are not included in the latest release. Refer https://github.com/microsoft/LightGBM/pull/3299#issuecomment-772614889 to for workaround.

**Comment by jameslamb** [comment]
@flippercy by the way, linear trees are available in the R package. Just not released to CRAN yet.

You can see https://github.com/microsoft/LightGBM/blob/fc2da2ec84e2dd9aa06421edcb5a625808eb546e/R-package/tests/testthat/test_basic.R#L348 for a usage example, and https://github.com/microsoft/LightGBM/blob/master/R-package/README.md#installation for details on how to build the R package from source.

In the future, if you find yourself turning to `{reticulate}` because the R package is missing something, could you please open a feature request here? It is hard for us to keep up with the full list of differences between R and Python, so that kind of feature request is very helpful.

**Comment by flippercy** [comment]
Thank you both! I will try the methods you suggested while waiting for the next official release!

**Comment by flippercy** [comment]
Just to confirm - does monotone_constraints work with linear_tree? (I believe so but want to double check).

Thank you!

**Comment by jameslamb** [comment]
I think @btrotta is better-qualified to answer that question than I am.

**Comment by btrotta** [comment]
@flippercy It is not possible to enforce monotone constraints when using linear tree. This is because the linear models are fitted on each leaf after the whole tree is constructed, and we do not take monotone constraints into account when fitting the linear models.

However, you can still use monotone constraints when linear_tree is true, and the constraints will be taken into account when choosing the tree split points. As noted above, in general the resulting linear tree will not be monotone, but it will probably be "closer to" monotone than it would be otherwise.

**Comment by flippercy** [comment]
@btrotta thank you for the response!

@jameslamb can someone add this to the documentation of lightGBM parameters at <https://lightgbm.readthedocs.io/en/latest/Parameters.html>, please?

Thank you!
